### PR TITLE
WIP example: application fix

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
- Fixes the example application generating the correct links.
- Updates the link generation for files download.
- Updates to Flask>=0.11
- Adds support for webasserts>=0.12
- Adds the shebang to `run-tests.sh`

Signed-off-by: Javier Delgado javier.delgado.fernandez@cern.ch

### CODE REVIEW

File: invenio_previewer/templates/invenio_previewer/macros.html [hunk 1]
```diff
@@ -54,7 +54,7 @@
         <td class="nowrap">{{ file.size }}</td>
         <td class="nowrap">
           <span class="pull-right">
-            <a class="btn btn-xs btn-default" href="{{ file.uri }}">
+            <a class="btn btn-xs btn-default" href="{{ url_for('invenio_records_ui.recid_download', pid_value=file.pid.pid_value, filename=file.file.key) }}">
```
Comment (JavierDelgadoFernandez) [inline invenio_previewer/templates/invenio_previewer/macros.html]: @lnielsen The links for the file download are a bit hardcoded right now, but the endpoint used in each installation could be different. Previously we were using ``files.uri`` but it is not possible now. How can I calculate it?

File: examples/app.py [hunk 2]
```diff
@@ -169,17 +176,19 @@ def create_object(bucket, file_name, stream):
     provider = RecordIdProvider.create(object_type='rec', object_uuid=rec_uuid)
     data = {
         'pid_value': provider.pid.pid_value,
-        'files': [
+        '_files': [
```
Comment (lnielsen) [inline examples/app.py]: @JavierDelgadoFernandez Note that @sourabhlal is updating this section to use Invenio-Records-Files. @sourabhlal any updates on when you'll updated PR #40

File: invenio_previewer/templates/invenio_previewer/macros.html (hunk 1 above)
Comment (lnielsen) [inline invenio_previewer/templates/invenio_previewer/macros.html]: I would do it similar to the ``preview_file`` macro:

```
{%- macro file_list(download_endpoint, pid_value, files) %}
{% for file in files %}
{{ url_for(download_endpoint, pid_value=pid_value, filename=f.key)}}
{%- endear %}
{%- endmacro %}
```

Comment (lnielsen) [inline invenio_previewer/templates/invenio_previewer/macros.html]: Can you make the file listing match this:

screen shot 2016-08-31 at 08 40 53

https://github.com/zenodo/zenodo/blob/master/zenodo/modules/records/templates/zenodo_records/box/files.html#L46-L65

Comment (JavierDelgadoFernandez) [inline invenio_previewer/templates/invenio_previewer/macros.html]: Thanks a lot @lnielsen ;)

However, the CSV previewer has the same issue. I will comment it in another message to show the line.

File: invenio_previewer/templates/invenio_previewer/csv_bar.html [hunk 3]
```diff
@@ -34,7 +34,7 @@
 {%- endblock %}
 
 {% block panel %}
-  <div id="d3_csv_previewer" data-csv-source="{{ file.uri }}"
+  <div id="d3_csv_previewer" data-csv-source="{{ url_for('invenio_records_ui.recid_download', pid_value=file.pid.pid_value, filename=file.file.key) }}"
```
Comment (JavierDelgadoFernandez) [inline invenio_previewer/templates/invenio_previewer/csv_bar.html]: @lnielsen Here it is the same issue with the URL generation. I can look for another way to send back the document, but the problem will be there for the PNG and JPG images previewer.

File: examples/app.py (hunk 2 above)
Comment (sourabhlal) [inline examples/app.py]: Its done now.

File: examples/app.py [hunk 4]
```diff
@@ -93,7 +94,7 @@
 
 
 10. Open a web browser and enter to the url
-`http://localhost:5000/records/RECORD_PID/preview` where
+`http://localhost:5000/records/RECORD_PID/preview/FILENAME` where
```
Comment (alizeepace) [inline examples/app.py]: Not sure, but shouldn't it be RECORD_ID ?

Comment (jirikuncar) [inline examples/app.py]: or better `<pid_value>` as it's defined in the route

**Comment by lnielsen** [comment]
Superseded

**Pull request closed without merging**